src/Endatix.Core/Entities/Submission.cs

```diff
@@ -22,6 +21,15 @@ public Submission(string jsonData, long formId, long formDefinitionId, bool isCo
         SetCompletionStatus(isComplete);
     }
 
+    public Submission(long id, string jsonData, long formId, long formDefinitionId, bool isComplete, int currentPage, string? metadata, 
```

Comment:
I am for merging this fast if needed, but please do check my recommendation for using specification with entity projection to a DTO.

Reply:
Now I use SubmissionDto and it works without the need of changes in Submission entity class. Thanks for the suggestion.
We need to clarify what types of objects (Entity objects, API models, DTOs) the different endpoints return because now it's inconsistent.